src/probability/extrealScript.sml

```diff
@@ -1133,28 +1178,38 @@ val extreal_sub_add = store_thm
     rpt Cases
  >> RW_TAC std_ss [extreal_ainv_def, extreal_sub_def, extreal_add_def, real_sub]);
 
-val sub_0 = store_thm
-  ("sub_0", ``!x y. (x - y = 0) ==> (x = y)``,
+Theorem sub_0 :
+    !x y :extreal. (x - y = 0) ==> (x = y)
+Proof
     rpt Cases
  >> RW_TAC std_ss [extreal_sub_def, num_not_infty, extreal_of_num_def, extreal_11,
-                   REAL_SUB_0]);
+                   REAL_SUB_0]
+QED
```

Comment:
Can't this be an iff (and a [simp])?

Reply:
No, `sub_0` cannot be an iff theorem without adding more antecedents: if `x = y = PosInf` or `x = y = NegInf`, then `x - y` is not specified, thus `x - y = 0` doesn't hold ...

Reply:
OK - that's fair.  Thanks.